Changed example text for adding a custom repo (bug 1061601)

lproven opened the pull request:
Example changed to the one [given in Bugzilla](https://bugzilla.suse.com/show_bug.cgi?id=1061601)

lproven commented:
On Thu, 21 Dec 2017 01:26:40 -0800
Christoph Wickert <notifications@github.com> wrote:

> cwickert requested changes on this pull request.
> 
> I think we can make the URL even shorter:
> http://my.example.com/repo/... instead of
> http://my.example.com/CAASP/repo/...

Done.

As per the suggestion on Github, I have removed the code comments from the zypp_repos example, and inserted a section explaining all the available options.

There is some merge problem on Github that I do not (yet) understand, so I request that you take a look at this. I welcome instruction on how to fix it.


> The rest looks good to me, but you really need to fix your editor to not reformat text all the time.

I am exploring options...

-- 
Liam Proven - Technical Writer, SUSE Linux s.r.o.
Corso II, Křižíkova 148/34, 186-00 Praha 8 - Karlín, Czechia
Email: lproven@suse.com - Office telephone: +420 284 241 084

lproven commented:
I have moved the </para> tag to before <variablelist> which fixes the validation error.

However, the file has been reflowed, again. I don't know why. 

I cannot see any difference at all in the section highlighted as a merge conflict, so I have simply deleted 1 copy of the text.

This should have resolved both the errors.

cwickert commented on xml/deployment_configuration.xml:
```diff
@@ -129,7 +129,9 @@
     time and must be in the format:
-    <literal>cloud-init-url=http://<replaceable>hostname.domain</replaceable>/cloud.cfg</literal>.
+    <literal>
+     cloud-init-url=http://<replaceable>hostname.domain</replaceable>/cloud.cfg
```
This is still pending, please fix it if you want me to approve this PR.

cwickert commented on xml/deployment_configuration.xml:
```diff
@@ -318,15 +320,108 @@ ssh_authorized_keys:
     </para>
    </sect3>
-   <sect3 xml:id="user-data.config.adding.custom.repository">
+      <sect3 xml:id="user-data.config.adding.custom.repository">
```
Still pending…

cwickert commented on xml/deployment_configuration.xml:
```diff
@@ -318,15 +320,108 @@ ssh_authorized_keys:
-<screen>zypp_repos:
-  "Update Repo":
-      baseurl: http://my.example.com/CAASP/repo/SUSE-CAASP-1.0-CUSTOM/</screen>
+    <para> You can add a custom software repository to your system by using the
```
Still pending…

cwickert commented on xml/deployment_configuration.xml:
```diff
@@ -318,15 +320,109 @@ ssh_authorized_keys:
+            - id: opensuse-oss-update
+              name: os-oss-up
+              baseurl: http://my.example.com/repo/SUSE-CAASP-&productnumber;-CUSTOM/update
+      </screen>
```
The indention of the &lt;/screen&gt; tag is not correct, it should match the opening tag. Because of the 1:1 display, you might want to not indent these two tags *at all* (at least this is what daps-xmlformat does).

cwickert commented on xml/deployment_configuration.xml:
```diff
@@ -318,15 +320,109 @@ ssh_authorized_keys:
+    <para> You can add a custom software repository to your system by using the
+      <literal>zypp_repos</literal> option: </para>
+    <screen>
+        zypper:
```
Note that the content of the screen element is displayed 'as-is' including whitespaces, any indention in the source are present in the printed media. Don't indent something that shouldn't be. In this case, zypper shouldn't be indented, only the following lines should be relative to it.

cwickert replied:
tl;dr: Unindent everything by 4 chars.

cwickert commented on xml/deployment_configuration.xml:
```diff
@@ -318,15 +320,109 @@ ssh_authorized_keys:
+              name: os-oss-up
+              baseurl: http://my.example.com/repo/SUSE-CAASP-&productnumber;-CUSTOM/update
+      </screen>
+    <para>The options available are: </para>
```
Line breaks, but no trailing whitespace please…

cwickert commented on xml/deployment_configuration.xml:
```diff
@@ -318,15 +320,109 @@ ssh_authorized_keys:
+         <literal>99</literal>. </para>
+       </listitem>
+      </varlistentry>
+     </variablelist>
```
The content of this paragraph is very good, but I suggest you start of with the mandatory options (id, nam, and baseurl) and then list the optional ones.

And then there are the usual coding style problems (indention, line breaks)

lproven commented:
On Wed, 03 Jan 2018 11:28:18 +0000 (UTC)
Christoph Wickert <notifications@github.com> wrote:

> I really like the new content, but the formatting needs work.

Thank you! I am glad to hear that.

I _think_ I have fixed all the formatting issues now. I have reformatted the file with ``daps-xmlformat -i'' and then hand-fixed the new section:
* removed leading and terminating spaces inside <para>s
* removed indent from <screen>

I have split <para></para> if there are multiple lines of text between them, and kept it consistent for all entries in a single list.

I have kept <term></term> on the same line.

DAPS has slightly changed the layout of some lines after the new section, but only the formatting, nothing else.

-- 
Liam Proven - Technical Writer, SUSE Linux s.r.o.
Corso II, Křižíkova 148/34, 186-00 Praha 8 - Karlín, Czechia
Email: lproven@suse.com - Office telephone: +420 284 241 084

lproven commented:
The relevant changes are now in the ``develop'' branch. Closing this PR.
